File: server/common/protocol/ack_message.py
from common.protocol.message import Message


class AckMessage(Message):
    ACK_OP = bytes([0x02])
    
    def serialize(self) -> bytes:
        return self._build_header(self.ACK_OP, 0)
    
    def should_ack(self) -> bool:
        return False

File: client/common/protocol/ack_message.go
package protocol

import "errors"

type Ack struct{}

func (ack *Ack) Serialize() []byte {
	return buildHeader([]byte{ACK_OP}, 0)
}

func (ack *Ack) Deserialize(stream []byte) error {

	if !compareStreams(ack.Serialize(), stream) {
		return errors.New("invalid Ack message")
	}
	return nil
}

func (ack *Ack) ShouldAck() bool {
	return false
}
